docs/docs/guides/atom-effects.md

```diff
@@ -239,6 +241,80 @@ const currentUserIDState = atom({
 });
 ```
 
+### Asynchronous
+
+If your persisted data needs to be `await`ed, you can either create an `asynchronous` function and call that or use a `promise` in the setSelf function 
+
+- Using the promise approach you'll be able to wrap the components inside of the `<RecoilRoot/>` with a `<Suspense/>` component to show a fallback state while waiting for `Recoil` to load the persisted values.
+- This is not possible with the async await approach.
+
+#### Promise
+
+```jsx
+const localStorageEffect = key => ({setSelf, onSet}) => {
+  setSelf(
+    localStorage.getItem(key).then((savedValue) => {
+      if (savedValue != null) {
+        return JSON.parse(savedValue);
+      }
+    }),
+  );
+
+  onSet(newValue => {
+    if (newValue instanceof DefaultValue) {
+      localStorage.removeItem(key);
+    } else {
+      localStorage.setItem(key, JSON.stringify(newValue));
+    }
+  });
+};
+
+const currentUserIDState = atom({
+  key: 'CurrentUserID',
+  default: 1,
+  effects_UNSTABLE: [
+    localStorageEffect('current_user'),
+  ]
+});
+```
+
+
+#### Async Await
+
```

Comment:
Clarify that with this approach, in contrast to the above, that it will use the default value until the async value was set.

Reply:
I clarified the pros/cons up at the asynchronous title, so you can read the difference together to make it easier to compare.

https://github.com/facebookexperimental/Recoil/pull/828/files#diff-33b7a54bcbd3c5dce0fc9f70d50de2e2a9260d9865c0a890252b2d92bb9d668eR248-R249

I can move each part down under each title as well, but I figure maybe it was unnecessary repetition.

Reply:
After reading through again I think it would be clearer to explain each approach's quirks in the sections themselves, thanks.

Reply:
@drarmstr That's fine by me.
I'll move each part down :)